vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Binary caching troubleshooting guide
+description: Troubleshooting your binary caching setup
+author: dan-shaw
+ms.author: danshaw2
+ms.date: 9/20/2023
+ms.prod: vcpkg
+---
+
+# Binary caching troubleshooting guide
+
+This guide is intended for vcpkg binary caching users that are running into unexpected behavior from this experience. Please review the [binary caching](./binarycaching.md) reference documentation for a basic understanding of what this feature does and how it works.
+
+## Set up vcpkg to print debugging information
+
+Debug mode outputs information such as the compiler verison and ABI hashes, which will help us to diagnose any binary caching issues.
+
+* [Classic mode](./classic-mode.md): add `--debug` to your command invocation.
+* CMake toolchain: set `-DVCPKG_INSTALL_OPTIONS="--debug"` in your `cmake` command call or in your `CMakePresets.json` file.
+* MSBuild/Visual Studio: set the property `VcpkgAdditionalInstallOptions` to `--debug`
+
+## Troubleshooting binary caching configuration issues
+
+There are many choices for a binary caching provider including AWS, NuGet, Azure. While each provider will have its own error messages, we expect the type of error and the actual fix will be the same across all providers. We will walk through a specific example of using NuGet as a binary cache provider.
+
+### Binary package ABI
```

Comment:
This is helping the user understand our debug output. It should merge into the debugging information section instead of being in the "Problem -> Solution" section.

Reply:
I've restructured the entire page based on what we discussed offline.